### ISSUE TITLE
PEcAN errors with species IDs

### ISSUE THREAD

**Issue by SimonGoring** [PalEON-Project/PalEON-FIA#9]
Currently, running the FIA code through Pecan is returning errors that I can't interpret:

```
 Error in attr(data, "mcpar") <- c(start, end, thin) : 
  attempt to set an attribute on NULL 
```

Several groups are also returning:

```
"READ.ALLOM.DATA: ** No allometric data available **"
```

My assumption is that these are species grouping that don't have allometric data, or that this plays into it, but the output is hard to interpret (for the - apparently) third pft:

```
[[3]]
[1] 2

[[1]]
[1] "READ.ALLOM.DATA: ** Warning no match of PFT to tally data **"

$acronym
[1] "CELTI" "CELA"  "CELAR" "CEOC" 

$spcd
[1] 460 461 463 462

[1] "READ.ALLOM.DATA: ** No allometric data available **"
$acronym
[1] "CHTH2" "JUSP"  "JUAS"  "JUSC"  "JUVI"  "JUVIS" "TAAS"  "TADI2" "THOC2"

$spcd
[1]  43  57  61  66  68  67 222 221 241
```

@andydawson have you run into this?  Also tagging @paciorek @crollinson @mdietze 

The code should run as-is (except for breaking in the Pecan section).

**Comment by SimonGoring** [comment]
Note, this all refers to [PEcAn_biom.R](https://github.com/PalEON-Project/PalEON-FIA/blob/master/R_scripts/PEcAn_biom.R).

**Comment by SimonGoring** [comment]
This issue follows on from the point in commit ecc1fbeae1b18083c0a8eacbcf140b2d373696ca.  To figure this out, in part, I need to match taxa to the [Jenkins table](https://github.com/PecanProject/pecan/blob/master/modules/allometry/data/Table3_GTR-NE-319.v2.csv) in the Pecan `allometry` module.

We need to make a decision about how we assign to species here.  We are calculating above ground biomass (component 2 in the table linked above) for all species, but for some species (like balsam fir) we [have the explicit parameters in the table](https://github.com/PecanProject/pecan/blob/master/modules/allometry/data/Table3_GTR-NE-319.v2.csv#L49).  For others, we have to go to the higher level [Eastern Conifers](https://github.com/PecanProject/pecan/blob/master/modules/allometry/data/Table3_GTR-NE-319.v2.csv#L4), or [Eastern Hardwoods](https://github.com/PecanProject/pecan/blob/master/modules/allometry/data/Table3_GTR-NE-319.v2.csv#L2570).

I'm not sure why we haven't had this discussion, or if we had, I can't remember why we chose to use the PFT allocations.  Maybe, because of that complexity, we decided to go with the pre-assigned PFTs from Dietze & Moorcroft?

I think the discussion should be here, and then, I propose we move the answer to the wiki for this repository.

Adding @IceAgeEcologist to this discussion and @jpeters7 @davidjpmoore

**Comment by crollinson** [comment]
You're right that those errors are what you get when the species you're trying to pull don't have any allometry equations -- or at least none for the component you're trying to pull.  You'll either need to use "expert judgement" to find a similar species & recode it to pull that species (for example substituting Acer saccharum for Acer saccharinum is probably a safe bet) or you assign it to a PFT.  That PFT doesn't have to already be in the Jenkin's DB.  One nice thing about the PEcAn module is that you can basically make custom PFT allometry equations by defining a PFT that will pull and synthesize equations from a bunch of species.  Maybe you're already doing that, in which case PEcAn *should* just be omitting those species from the PFT-level equation it produces.

**Comment by SimonGoring** [comment]
I think the problem is that there are some new pseudo-PFTs (like hackberry) that contain species IDs without any allometries.  So even doing the aggregation isn't working, and we need to add in an spcd like 1000 (eastern hardwoods, I think, off the top of my head) to make it work.

**Comment by crollinson** [comment]
What I'm trying to say is that you can either do that, or use Pecan to create an equation that combines for example oaks, dogwoods, and willows and apply that to hackberry.  You can either use existing ecosystem model-based groupings to assign Celtis to a PFT or you can pick several species of similar growth form to leverage (including the generalized eastern hardwoods).

**Comment by crollinson** [comment]
I think the bottom line is that there's no way around recoding those species/groups like Hackberry into either a pre-defined PFT that already exists in PEcAn or something that pulls a suite of species to come up with a best-guess equation (with uncertainty).

**Comment by SimonGoring** [comment]
Thanks @crollinson, yep, I agree, I think I was hoping for (1) confirmation that that's the case, and (2) opinions about how best to proceed, mostly from you and @mdietze given that you all have more experience with the FIA data than I do.

**Comment by jpeters7** [comment]
Hey @SimonGoring, @crollinson, @paciorek, @mdietze - Jason and I talked.  He wants to get this moving. @SimonGoring you could send the list of species that we do not have allometry for. Then Jason could take the first crack at deciding what other species/taxa/PFT to use for the allometries. We will probably also reach out to Charlie and Steve for input as well.

**Comment by andydawson** [comment]
Has this been resolved @SimonGoring ?

**Comment by paciorek** [comment]
Charlie is going to check through the 'proposed' column from Simon and give his judgment of what Pecan-allometry taxon should be used for problematic FIA species.
